pytorch_lightning/accelerators/ddp_backend.py

```diff
@@ -127,14 +137,116 @@ def __ddp_script_mode_setup(self):
 
         self.task_idx = 0
 
+        # wait for all the procs to start
+        sleep(2)
+
     def train(self):
         model = self.trainer.model
         if self.mode == 'ddp':
-            results = self.ddp_train_tmp(process_idx=self.task_idx, mp_queue=None, model=model, is_master=True)
-            del os.environ['WORLD_SIZE']
+            results = self.ddp_train(process_idx=self.task_idx, mp_queue=None, model=model, is_master=True)
+            if 'WORLD_SIZE' in os.environ:
+                del os.environ['WORLD_SIZE']
```

Comment:
n00b question: should this happen in a teardown step?

Reply:
probably haha.

i'm not sure what will happen if we move it though. I want to lock down the tests then we can see where it goes